pkg/endpoint/connector/ipvlan.go

```diff
@@ -215,29 +215,32 @@ func getIpvlanMasterName() string {
 }
 
 // SetupIpvlan creates an ipvlan slave in L3 based on the master device.
-func SetupIpvlan(id string, mtu int, masterDev int, ep *models.EndpointChangeRequest) (*netlink.IPVlan, *netlink.Link, string, error) {
+func SetupIpvlan(id string, mtu, masterDev int, mode netlink.IPVlanMode, ep *models.EndpointChangeRequest) (*netlink.IPVlan, *netlink.Link, string, error) {
 	if id == "" {
 		return nil, nil, "", fmt.Errorf("invalid: empty ID")
 	}
 	if masterDev == 0 {
 		return nil, nil, "", fmt.Errorf("invalid: master device ifindex")
 	}
+	if mode != netlink.IPVLAN_MODE_L3 && mode != netlink.IPVLAN_MODE_L3S {
+		return nil, nil, "", fmt.Errorf("invalid: ipvlan operation mode")
```

Comment:
Maybe it would be good to report the invalid mode in the error msg?

Reply:
It's actually printed in cilium-cni.go which takes the opmode string and translates it to netlink.IPVLAN_MODE_* enum. Unsupported one will be dumped there, the one above is internal.